### ISSUE TITLE
selenium.common.exceptions.NoSuchElementException

### ISSUE THREAD

**Issue by xingzhehe** [Ed1123/visa_web_scraper#8]
Thanks a lot for the great project!
I just ran into some bug(?) here.
After I run the `python3 selenium_scraper.py`, it gives me the following `NoSuchElementException`. I have not got chance to type in anything yet. Do I need to change some part of the code ourselves?
Thanks in advance for sharing it!

> Checking for changes.
> Page reached.
> Traceback (most recent call last):
>   File "selenium_scraper.py", line 142, in <module>
>     main()
>   File "selenium_scraper.py", line 139, in main
>     run_visa_scraper(url, text)
>   File "selenium_scraper.py", line 107, in run_visa_scraper
>     if has_website_changed(driver, url, no_appointment_text):
>   File "selenium_scraper.py", line 76, in has_website_changed
>     main_page = driver.find_element(By.ID, 'main')
>   File "/Users/xxx/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 858, in find_element
>     'value': value})['value']
>   File "/Users/xxx/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 429, in execute
>     self.error_handler.check_response(response)
>   File "/Users/xxx/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 243, in check_response
>     raise exception_class(message, screen, stacktrace)
> selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":"[id="main"]"}

**Comment by Ed1123** [comment]
Judging by the tracelog it seems the website change some web element. It will require debugging, but I'm not able to reproduce it the error (since I don't have access to the appointment website anymore), nor I have much time to do it. Sorry about that.

**Comment by xingzhehe** [comment]
I see. Thanks anyway for this great project!

**Comment by Ed1123** [comment]
Thank you 🙂

**Comment by Ed1123** [comment]
Hey, I'm gonna work on improving it. Maybe creating another one. Just an update. 🙂

**Comment by optimisticinlimbo** [comment]
Hi Ed1123, any news on the update? :)

**Comment by Ed1123** [comment]
I've been exploring alternatives since updating it was gonna be same or more effort than creating a new one.
I saw [Automata](https://www.automa.site/) extension and gave it a try, but couldn't make it work.
I also saw [Puppetter](https://pptr.dev/). Haven't try it yet, but may be worth taking a look. One thing is sure, rewriting won't be in Python. Probably Go, but may explore something else.

I'm gonna close this ticket, but feel free to open a new one or contact me on LinkedIn. :)
